spec/support/matchers/be_rendered.rb

```diff
@@ -0,0 +1,21 @@
+RSpec::Matchers.define :be_rendered do |with: nil|
+  match do |page|
+    if with.nil?
+      !page.native.inner_html.empty?
+    else
+      page.has_css?("body") && page.find("body").native.inner_html == with
+    end
+  end
+
+  failure_message do |page|
+    if page.has_css?("body")
+      "expected page to be rendered with #{with}, but was rendered with #{page.find("body").native.inner_html}"
+    else
+      "expected page to be rendered with #{with}, but was not rendered"
+    end
+  end
+
+  failure_message_when_negated do |page|
+    "expected page not to be rendered, but was rendered with #{page.native.inner_html}"
```

Comment:
In order to have more concise error messages in our usual scenario, how about:

```suggestion
    if page.has_css?("body")
      "expected page not to be rendered, but was rendered with #{page.find("body").native.inner_html}"
    else
      "expected page not to be rendered, but was rendered with #{page.native.inner_html}"
    end
```

Maybe the `else` part isn't strictly necessary; it's there just because there might be edge cases we haven't considered and because without it the `!page.native.inner_html.empty?` condition used in the matcher seems a bit strange :thinking:.